src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -497,7 +497,7 @@ bool usesOriginalInput(BoundDecisionDagNode node)
                 {
                     var field = loweredInput.Type.TupleElements[i].CorrespondingTupleField;
                     Debug.Assert(field != null);
-                    var expr = loweredInput.Arguments[i];
+                    var expr = getExpression(i, loweredInput);
```

Comment:
Here, we can't just access Arguments[i] for two reasons:

1. If i == 8, the result expression will be a tuple, this is not intended.
2. If i > 8, this will throw an exception due to the other fix I made.